### PR TITLE
CSP-484-GETPartnerMember

### PR DESCRIPTION
**Pull request by ravihejmadi**
Added Queries & Tests
Modified Controller & Test

### CODE REVIEW

File: src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberResult.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using SFA.DAS.AANHub.Domain.Entities;
+
+namespace SFA.DAS.AANHub.Application.Partners.Queries
+{
+    public class GetPartnerMemberResult
+    {
+        public Guid MemberId { get; set; }
+        public string? Email { get; set; }
```
Comment (LiamWright) [inline src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberResult.cs:8]: I've just got a similar comment on my recent PR, but make these non-nullable and just initialise the property with ` = null!;`

File: src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberResult.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        public string? Organisation { get; set; }
+
+        public static implicit operator GetPartnerMemberResult?(Partner partner)
```
Comment (LiamWright) [inline src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberResult.cs:11]: Partner should be nullable rather than `GetPartnerMemberResult`

File: src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberQueryValidator.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿using FluentValidation;
+
+namespace SFA.DAS.AANHub.Application.Partners.Queries
+{
+    public class GetPartnerMemberQueryValidator : AbstractValidator<GetPartnerMemberQuery>
+    {
+        public GetPartnerMemberQueryValidator()
+        {
+            RuleFor(a => a.UserName)
+                .NotNull()
```
Comment (LiamWright) [inline src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberQueryValidator.cs:10]: Mirroring another comment that I got on my GET PR. `NotEmpty` covers `NotNull` already

File: src/SFA.DAS.AANHub.Application.UnitTests/Partners/Queries/GetPartnerMemberResultTest.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    public class GetPartnerMemberResultTest
+    {
+        [Test, AutoMoqData]
+        public void Partner_PopulatesGetPartnerMemberResultFromPartner(Partner partner)
+        {
+            var response = (GetPartnerMemberResult?)partner;
+
+            Assert.That(response, Is.Not.Null);
+
+            Assert.AreEqual(partner.MemberId, response!.MemberId);
```
Comment (LiamWright) [inline src/SFA.DAS.AANHub.Application.UnitTests/Partners/Queries/GetPartnerMemberResultTest.cs:17]: We should probably agree to whether we use FluentAssertions or NUnit's `Assert` syntax to validate tests.

Thoughts @DevenKShah ?

File: src/SFA.DAS.AANHub.Application.UnitTests/Partners/Queries/GetPartnerMemberQueryHandlerTests.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+            Guid memberid = new();
+
+            var partnersReadRepositoryMock = new Mock<IPartnersReadRepository>();
+            var partner = new Partner();
+            partnersReadRepositoryMock.Setup(a => a.GetPartnerByUserName(userName)).ReturnsAsync(partner);
+            var sut = new GetPartnerMemberQueryHandler(partnersReadRepositoryMock.Object);
+
+            var result = await sut.Handle(new GetPartnerMemberQuery(userName), new CancellationToken());
+
+            Assert.AreEqual(memberid, result!.Result.MemberId);
```
Comment (LiamWright) [inline src/SFA.DAS.AANHub.Application.UnitTests/Partners/Queries/GetPartnerMemberQueryHandlerTests.cs:26]: Probably should just check that the other properties are populated as expected

File: src/SFA.DAS.AANHub.Application.UnitTests/Partners/Queries/GetPartnerMemberQueryValidatorTests.cs [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+using SFA.DAS.AANHub.Application.Partners.Queries;
+
+namespace SFA.DAS.AANHub.Application.UnitTests.Partners.Queries
+{
+    [TestFixture]
+    public class GetPartnerMemberQueryValidatorTests
+    {
+        [TestCase("username", true)]
+        [TestCase(null, false)]
+        [TestCase("", false)]
```
Comment (LiamWright) [inline src/SFA.DAS.AANHub.Application.UnitTests/Partners/Queries/GetPartnerMemberQueryValidatorTests.cs:12]: Mirroring feedback once more but I'd add a test for whitespace, just in case
```suggestion
        [TestCase(" ", false)]
```

File: src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberResult.cs (hunk 2 above)
Comment (ravihejmadi) [reply src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberResult.cs:11]: Afraid not as null is passed to GetPartnerMemberResult (line 14). The code is very similar to the [GETAdmin endpoint](https://github.com/SkillsFundingAgency/das-aan-hub-api/pull/44/files#diff-2ce896ea44c5aadf66bc1f10033e2f51f33acf97af157a7782467aafe282f89a)

File: src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberQueryValidator.cs (hunk 3 above)
Comment (ravihejmadi) [reply src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberQueryValidator.cs:10]: Fixed. Thanks for the tip!

File: src/SFA.DAS.AANHub.Application.UnitTests/Partners/Queries/GetPartnerMemberResultTest.cs (hunk 4 above)
Comment (ravihejmadi) [reply src/SFA.DAS.AANHub.Application.UnitTests/Partners/Queries/GetPartnerMemberResultTest.cs:17]: I will leave this for us to look at a later stage

File: src/SFA.DAS.AANHub.Application.UnitTests/Partners/Queries/GetPartnerMemberQueryHandlerTests.cs (hunk 5 above)
Comment (ravihejmadi) [reply src/SFA.DAS.AANHub.Application.UnitTests/Partners/Queries/GetPartnerMemberQueryHandlerTests.cs:26]: Added other properties. Thanks for this.

File: src/SFA.DAS.AANHub.Application.UnitTests/Partners/Queries/GetPartnerMemberQueryValidatorTests.cs (hunk 6 above)
Comment (ravihejmadi) [reply src/SFA.DAS.AANHub.Application.UnitTests/Partners/Queries/GetPartnerMemberQueryValidatorTests.cs:12]: Applied recommendation

File: src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberResult.cs (hunk 1 above)
Comment (ravihejmadi) [reply src/SFA.DAS.AANHub.Application/Partners/Queries/GetPartnerMemberResult.cs]: Fixed

File: src/SFA.DAS.AANHub.Application.UnitTests/Partners/Queries/GetPartnerMemberResultTest.cs (hunk 4 above)
Comment (DevenKShah) [reply src/SFA.DAS.AANHub.Application.UnitTests/Partners/Queries/GetPartnerMemberResultTest.cs:17]: I have not been consistent myself when it comes to assertion. I don't mind what you use, as both libs are reference anyway.

**Review by LiamWright — DISMISSED** [review]

**Review by LiamWright — APPROVED** [review]

**Comment by ravihejmadi** [comment]
Tested successfully in Test & PO Review complete
